sync_issues_to_jira/push_event.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+#
+# Copyright 2019 Espressif Systems (Shanghai) PTE LTD
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from github import Github
+import os
+import re
+
+def handle_push_event(event):
+    issue_numbers = []
+    for commit in event['commits']:
+        commit_message = commit['message']
+        issue_numbers += parse_commit_message(commit_message)
+    
+    github = Github(os.environ['GITHUB_TOKEN'])
+    repo = github.get_repo(os.environ['GITHUB_REPOSITORY'])
+    for issue in issue_numbers:
+        gh_issue = repo.get_issue(int(issue))
+        if gh_issue.pull_request:
+            update_pull_request(gh_issue.as_pull_request())
+
+def update_pull_request(pull_request):
+    print("Updating %s" % pull_request)
+    if pull_request.state == 'open':
+        print('Pull request is open, nothing to update.')
+        return
+    original_title = pull_request.title
+    # Prepend [Merged] to the pull request title
+    new_title = '[Merged] ' + original_title
+    pull_request.edit(title=new_title)
+    # Thank contributor for opening pull request. Let them know we didn't throw it away
+    pull_request.create_issue_comment('The pull request has been cherry-picked, the commit is linked above.\
+        Thank you for your contribution!')
+
+
+def parse_commit_message(commit_message):
+    # Regex matches numbers that come after Fix, fix, Fixed, fixed, Fixes, fixes keyword followed by any
+    # combination of spaces and colons, followed by exactly one hashtag. The same applies for Close and Resolve
+    # keywords and their combinations. Note: fixing, closing and resolving don't work.
+    # Only first number is picked. To match multiple numbers you have to add fix or close or resolve or implement keyword
+    # for each of them.
+    # Example:
+    # fixed: #1 #2 #3
+    # resolved   ::: ::: :: #4
+    # closes: ##5
+    # fix: # 6
+    # asdfresolves #7
+    # closeasdf: #8
+    # closes #9 <any sting in between> fixed #10
+    # fixing #11
+    # Matches: [1, 4, 7, 9, 10]
+    pattern = re.compile('(?:[Ff]ix(?:e[sd]?)(?:\ |:)+|(?:[Cc]los(?:e[sd]?)(?:\ |:)+)|(?:[Rr]esolv(?:e[sd]?)(?:\ |:)+))#(\d+)')
```

Comment:
in IDF we encourage developers to use the full GitHub URL here, to minimize the conflicts between GitLab which uses the same #-syntax for issues.

GitHub renders these the same as if the URL was not there, but if you look at the "raw" commit message then it is.

Example: https://github.com/espressif/esp-idf/commit/bdea007a379b272c37cf14c8cde493a26491f18b
but the actual commit message is  https://github.com/espressif/esp-idf/commit/bdea007a379b272c37cf14c8cde493a26491f18b.patch

Reply:
Also, a tip: you've written a possibly very efficient but also quite hard to debug regex here. A more human-friendly alternative for long term maintenance might be something like:

```
re.compile(r'(?:closes?|fixes|fix|resolves?|resolved)\s+#(\d+)', re.IGNORECASE)
```

(Even `closes?` may be unnecessarily clever when `close|closes` works just as well.)

As well as testing these locally in the python shell, there are some nice online tools like https://pythex.org/ that can help you out.

(Don't forget to use r-string form for Python, to avoid backslash escape misery.)

Reply:
Thank you for suggestions! I have fixed this in https://github.com/espressif/github-actions/pull/14/commits/513c1d25204a8f5d6530e4cbc901f79b388373b8